packages/components/src/custom-select-control-v2/styles.ts

```diff
@@ -105,10 +105,21 @@ export const Select = styled( Ariakit.Select, {
 } );
 
 export const SelectPopover = styled( Ariakit.SelectPopover )`
+	display: flex;
+	flex-direction: column;
+
+	background-color: ${ COLORS.theme.background };
 	border-radius: 2px;
-	background: ${ COLORS.theme.background };
 	border: 1px solid ${ COLORS.theme.foreground };
 
+	/* Same as popover component */
+	/* TODO: is there a way to read the sass variable? */
+	z-index: 1000000;
+
+	max-height: min( var( --popover-available-height, 400px ), 400px );
```

Comment:
This is fine for legacy (and for the purposes of this PR), but I think we will also need a way for consumers to override this (#49110).

Reply:
That's a good point — not necessarily part of the tweaks to V2 legacy, but important to keep in mind for future work. It's also tracked in #55023